src/nvim/ex_getln.c

```diff
@@ -4641,7 +4638,6 @@ int del_history_idx(int histype, int idx)
     history[histype][i] = history[histype][j];
     i = j;
   }
-  clear_hist_entry(&history[histype][i]);
```

Comment:
Absense of this leads to double-free when using `histdel()` (or any other way to clear history). Fix and test are in the third commit as well.